ludwig/models/predictor.py

```diff
@@ -317,7 +317,17 @@ def save_prediction_outputs(
     backend,
 ):
     postprocessed_output, column_shapes = flatten_df(postprocessed_output, backend)
-    postprocessed_output.to_parquet(os.path.join(output_directory, PREDICTIONS_PARQUET_FILE_NAME))
+
+    schema = "infer"
+    try:
+        # Schema is needed for Dask 2022.6+ otherwise probabilities columns will be inferred as strings
+        import pyarrow as pa
+
+        schema = pa.schema({feature + "_probabilities": pa.list_(pa.float64()) for feature in output_features})
+    except ImportError:
+        logger.warning("Could not import pyarrow in save_prediction_outputs()")
+
+    postprocessed_output.to_parquet(os.path.join(output_directory, PREDICTIONS_PARQUET_FILE_NAME), schema=schema)
```

Comment:
@hungcs Is the `to_parquet()` for Dask? What if we move this piece of code into https://github.com/ludwig-ai/ludwig/blob/master/ludwig/data/dataframe/dask.py? I think it would make sense to do schema inference internally within the function.

It's super interesting you're having to do this, because I actually saw a similar issue a while ago and tried to create a fix for schema inference myself: https://github.com/ludwig-ai/ludwig/pull/2318/files

Seems like maybe we should consolidate these and then just make this something we do for all dask dataframe creation related operations so that we don't allow dask to infer by itself?

cc: @tgaddair

Reply:
yeah this is for `dask to_parquet`

in the file you linked, we are doing `schema="infer"` (https://github.com/ludwig-ai/ludwig/blob/master/ludwig/data/dataframe/dask.py#L194), but we can change it to `schema=None` if we want

Reply:
I had a quick chat with some folks, and I think this change is good for now and you should proceed with it 👍 If we find more places in the future where we need to pass in custom schema objects to Dask, we can revisit whether we should just add that in everywhere within our custom DaskEngine function wrappers around Dask native function calls.

Reply:
@arnavgarg1 it looks like the unit tests are failing because we can't just provide a partial schema with one column - pyarrow will ignore all the columns that were not provided: https://github.com/apache/arrow/issues/14025

how do you feel about doing the original change:
```
 postprocessed_output.to_parquet(os.path.join(output_directory, PREDICTIONS_PARQUET_FILE_NAME),
                                        schema=None)
```

this is the behavior in dask 2022.5

Reply:
@hungcs That's correct, partial schemas won't work. Does passing in schema=None fix the issue that you're running into? Is it not the case that schema=None is that same as asking Dask to infer the schema?

Reply:
So, the downside of `schema=None` is that there is no checking that the schemas are consistent across partitions. So I guess we can run into issues with things like nullable dtypes etc downstream that Dask may complain about. 

Do you know what we end up doing with `postprocessed_output` after we write it as a parquet file? I took a quick look and it just seems like we're writing the predictions to disk and then not doing much else with it. I think since we're only doing this while writing to parquet hopefully it won't cause problems But if we are actively reading it back and using it for other things downstream like visualizations, it can potentially cause issues.

I think this is okay for now though 👍